src/Notmuch/Binding.chs

```diff
@@ -356,10 +358,19 @@ thread_get_messages ptr = liftIO $ withThread ptr $ \ptr' ->
     >>= messagesToList . Messages
 
 -- notmuch_thread_get_matched_messages -> Int
--- notmuch_thread_get_authors -> String
--- notmuch_thread_get_subject
+
+thread_get_authors :: Thread a -> IO (Maybe B.ByteString)
+thread_get_authors ptr = withThread ptr $ \ptr' -> do
+  r <- {#call thread_get_authors #} ptr'
+  if r == nullPtr
+     then pure Nothing
+     else Just <$> B.packCString r
+
+thread_get_subject :: Thread a -> IO B.ByteString
+thread_get_subject ptr = withThread ptr ({#call thread_get_subject #} >=> B.packCString)
```

Comment:
Hypothetical question... what happens with empty/missing subject header?

Reply:
yeah tried it out... empty string. Although checking the documentation again it says:

```C
/**                                                                                                                                                                                            
 * Get the subject of 'thread' as a UTF-8 string.                                                                                                                                              
 *                                                                                                                                                                                             
 * The subject is taken from the first message (according to the query                                                                                                                          * order---see notmuch_query_set_sort) in the query results that                                                                                                                               
 * belongs to this thread.                                                                                                                                                                     
 * [...]
 */
```